labs/sysdev-kernel-cross-compiling-stm32mp2/sysdev-kernel-cross-compiling-stm32mp2.tex

```diff
@@ -0,0 +1,141 @@
+\subchapter{Kernel - Cross-compiling}{Objective: Learn how to
+cross-compile a kernel for an ARM target platform.}
+
+After this lab, you will be able to:
+\begin{itemize}
+\item Checkout a stable version of the Linux kernel
+\item Set up a cross-compiling environment
+\item Cross compile the kernel for the \labboarddescription
+\item Use U-Boot to download the kernel
+\item Check that the kernel you compiled starts the system
+\end{itemize}
+
+\section{Setup}
+
+Stay in the \code{$HOME/__SESSION_NAME__-labs/kernel} directory.
+
+\section{Cross-compiling environment setup}
+
+To cross-compile Linux, you need to have a cross-compiling
+toolchain. We will use the cross-compiling toolchain that we
+previously produced, so we just need to make it available in the PATH:
+
+\begin{bashinput}
+$ export PATH=$HOME/x-tools/aarch64-training-linux-musl/bin:$PATH
+\end{bashinput}
+
+Also, don't forget to either:
+
+\begin{itemize}
+\item Define the value of the \code{ARCH} and \code{CROSS_COMPILE}
+  variables in your environment (using \code{export})
+\item {\bf Or} specify them on the command line at every invocation of
+  \code{make}, i.e.: \code{make ARCH=... CROSS_COMPILE=... <target>}
+\end{itemize}
+
+\section{Linux kernel configuration}
+
+By running \code{make help}, look for the proper Makefile target to
+configure the kernel for your processor.
+
+The standard configuration for this kernel is actually \code{defconfig},
+but this will generate a pretty big kernel with support for many other
+SoCs. However, we can reduce it to compile faster and get a small
+kernel.
```

Comment:
I'm not sure we should use those fragments which are very very ST-specific. What about doing what we do with the STM32MP1 lab, i.e use the "defconfig", and then tell people to disable platforms in menuconfig. This is going to be much more "generic" than using the ST-specific fragments.